globals.go

```diff
@@ -272,6 +275,38 @@ const (
 	InvalidDivideInputError    = "$divide only supports numeric types"
 )
 
+var TaskStatuses = []string{
+	TaskStarted,
+	TaskSucceeded,
+	TaskFailed,
+	TaskSystemFailed,
+	TaskTestTimedOut,
+	TaskSetupFailed,
+	TaskAborted,
+	TaskStatusBlocked,
+	TaskStatusPending,
+	TaskKnownIssue,
+	TaskSystemUnresponse,
+	TaskSystemTimedOut,
+	TaskTimedOut,
+}
+
+// FilterValidTaskStatuses returns a slice of task statuses that are valid and are searchable.
+// It returns an empty array if emptyArrayForAll is true
+func FilterValidTaskStatuses(statuses []string, emptyArrayForAll bool) []string {
+	var filteredStatuses []string
+	if emptyArrayForAll && utility.StringSliceContains(statuses, TaskAll) {
+		return []string{}
```

Comment:
Also I would preference initializing filteredStatuses as filteredStatuses := []string{} and just returning that here